### PR TITLE
Add support for generic Kubernetes client on e2e tests

### PR DESCRIPTION
**Pull request by TheSpiritXIII**
Minor refactoring. Makes the code more readable because:
- The namespace in the type definition is used.
- The code is shorter and easier to write.
- You don't need multiple clientsets.
- You can change types you create effortlessly.
- We can create a special Kubernetes client which adds subtest labels automagically.

### CODE REVIEW

File: e2e/operator_context_test.go [hunk 1]
```diff
@@ -337,46 +364,33 @@ func (tctx *OperatorContext) createBaseResources(ctx context.Context) error {
...
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		for k, v := range tctx.getSubTestLabels() {
+			labels[k] = v
+		}
+		obj.SetLabels(labels)
+
+		if err := kubeClient.Create(ctx, obj); err != nil {
+			return fmt.Errorf("create object %s: %w", string(doc), err)
```
Comment (bernot-dev) [inline e2e/operator_context_test.go:383]: Is `doc` the raw yaml? That's going to end up being an oddly formatted error, I think.

File: e2e/operator_context_test.go [hunk 2]
```diff
@@ -152,11 +163,12 @@ func newOperatorContext(t *testing.T) *OperatorContext {
 		T:              t,
 		namespace:      namespace,
 		pubNamespace:   pubNamespace,
+		kClient:        kClient,
 		kubeClient:     kubeClient,
```
[the remark this answers is not in the document]
Comment (TheSpiritXIII) [reply e2e/operator_context_test.go:167]: Future PRs will remove the `kubeClient` which should have been more accurately named `kubeClientset`.

File: e2e/operator_context_test.go [hunk 3]
```diff
@@ -257,33 +305,10 @@ func (tctx *OperatorContext) createBaseResources(ctx context.Context) error {
...
-			},
-			Data: map[string][]byte{
-				"key.json": b,
-			},
-		}
-		_, err = tctx.kubeClient.CoreV1().Secrets(tctx.pubNamespace).Create(ctx, secret, metav1.CreateOptions{})
-		if err != nil {
-			return fmt.Errorf("create GCP service account secret: %w", err)
-		}
+	}); err != nil {
```
[the remark this answers is not in the document]
Comment (TheSpiritXIII) [reply e2e/operator_context_test.go:308]: The Go style guide shows this as "Good" when it comes to inline struct initialization: https://google.github.io/styleguide/go/decisions#conditionals-and-loops

It's quite longer than the example on the guide but not long enough that it won't all fit on your screen.

[the remark this answers is not in the document]
Comment (bernot-dev) [reply e2e/operator_context_test.go:314]: Fair enough. Actually, I think I discussed pros and cons of a separate initialization vs inline literal with @bwplotka at one point, and we left it as a non-decision. I just know my preference. 😄

**Review by bernot-dev — APPROVED** [review]
LGTM

**Pull request merged**